Leanplum-SDK/Classes/LeanplumRequest.m

```diff
@@ -372,8 +374,11 @@ - (void)sendRequests:(BOOL)async
             dispatch_semaphore_signal(semaphore);
             LP_END_TRY
             
-            if (_response != nil) {
-                _response(operation, json);
+            if (_eventIndex >= requestsToSend.count) {
```

Comment:
may be more safe to move this logic to 
https://github.com/Leanplum/Leanplum-iOS-SDK/blob/develop/Leanplum-SDK/Classes/Leanplum.m#L884